### ISSUE TITLE
Some clarifications about method approximate_distribution

### ISSUE THREAD

**Issue by laurent-pincemaille** [MaartenGr/BERTopic#1082]
I will paraphrase the explications you gave in documentation to be sure to understand correctly the algorithm and then ask for the advantage of this way to do against other we could imagine. The goal is to give a kind of softclustering for the documents based on c-tfidf. 

So if we imagine one document "Solving the right problem is difficult." we can create a tokenset my_tokenset= ["solving the right", "the right problem", "right problem is", "problem is difficult"] of 4 sub-documents [let's say we already  computed the ctfidf matrix of the original corpus: let's say there is 3 topics and 1000 words in the vocabulary]. 
Then we can calculate a ctfidf-representation of each sub_doc  (wich will be really sparse since each sub-document do contain at max 3 words) and we compute the cosine similarity between theses 4 subdocs and the ctfidf matrix (original) wich will give 4 vector of size 3 (because we made the assumption of 3 topics). And finally we compute the sum of theses 4 vectors and finally normalize it to get a probability.

I found it quite complicate, why not taking the ctfidf-representation of the entire doc and just compute the cosine similarity with the ctfidf matrix (original) ?

**Comment by MaartenGr** [comment]
> I found it quite complicate, why not taking the ctfidf-representation of the entire doc and just compute the cosine similarity with the ctfidf matrix (original) ?

When you apply the cosine similarity of a document's c-TF-IDF and the topic's c-TF-IDF representation, what you will get out of that is a similarity matrix. This matrix will only tell you which documents are close to which topic and it will not tell you the distribution of topics in those documents. Just because topic 3 has a similarity of 0.6 with document 1 does not mean that document 1 consists of 60% of topic 3. 

In other words, the methodology above is meant to do two things. First, it creates a distribution of topics within a document to demonstrate which topics can be found within that document. Using a similarity matrix for that is much less reliable. Second, the topic distribution that you get also demonstrates where the distributions of those topics take place. More specifically, it shows which tokens and token sets belong to certain topics. It goes even as far as indicating that certain tokens can belong to multiple topics.

**Comment by laurent-pincemaille** [comment]
Very clear. Thanks a lot!
